pkg/utils/nodeutils.go

```diff
@@ -115,6 +121,79 @@ func IsMcfgPoolUsingKC(pool *mcfgv1.MachineConfigPool) (bool, string, error) {
 	return true, currentKCMC, nil
 }
 
+func AreKubeletConfigsRendered(pool *mcfgv1.MachineConfigPool, client runtimeclient.Client) (bool, error) {
+	// find out if pool is using a custom kubelet config
+	isUsingKC, currentKCMCName, err := IsMcfgPoolUsingKC(pool)
+	if err != nil {
+		return false, fmt.Errorf("failed to check if pool %s is using a custom kubelet config: %w", pool.Name, err)
+	}
+	if !isUsingKC || currentKCMCName == "" {
+		return true, nil
+	}
+	// if the pool is using a custom kubelet config, check if the kubelet config is rendered
+	kcmcfg := &mcfgv1.MachineConfig{}
+	err = client.Get(context.TODO(), types.NamespacedName{Name: currentKCMCName, Namespace: pool.Namespace}, kcmcfg)
```

Comment:
This is a nit, but a pool is not namespaced, but cluster-wide, so you can just remove the `Namespace` parameter and not use the `pool.Namespace` attribute, it will always be an empty string anyway.

Reply:
that's a good catch!